v3/src/utilities/js-utils.ts

```diff
@@ -1,6 +1,9 @@
-import { nanoid } from "nanoid"
+import { customAlphabet } from "nanoid"
 import { monotonicFactory } from "ulid"
 const ulid = monotonicFactory()
+// Use custom alphabet to avoid ambiguous characters, especially during mathematical formula evaluations.
+// By default, nanoid uses "-" sign, which is used in formulas for subtraction.
+const nanoid = customAlphabet("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz", 21)
```

Comment:
Note that the only change done here is removal of the minus sign `-`. nanoid would often generate attribute IDs like `"ATTRXYZ-ABC"`. And these attribute IDs are later used in the canonical formula that is parsed by Mathjs. So, mathjs obviously would interpret it as `ATTRXYZ - ABC` (ATTRXYZ minus ABC).

We could map raw attr IDs to some mathjs-safe-form, but since it's just one symbol causing issues, I think it makes more sense to remove it simply. No one really cares about the form of these IDs, so let's make them as convenient as possible.